### DISCUSSION TITLE
left join tables

### DISCUSSION THREAD

**Discussion by szanuje** [prisma/prisma#14262] [Q&A]
Hey, I'd like to query data with prisma, with the following scenario:

table A, has no unique constraints, always queried by column which might have duplicated keys (find many)
table B, has unique constraint

table A has keys that cannot be referenced in table B because such key does not exist, same with table B, so tables are separate subsets of data but has some common keys.

I'd like to perform a left join - `select * from table A left join table B on ...`
This will get all data from table A and join matching results from table B, the missing data will be filled with null.

I managed to do it by using `rawQuery` but I'd like to abstract it with prisma.

I tried to create relations but they always forced me to use foreign key or unique constraints but it's not possible here.

**Comment by austincrim** [comment]
Hey @szanuje 👋,

Could you provide an example schema so I can see if this is possible? At first glance, I don't think it is (without using `queryRaw`), but just wanted to make sure.

**Reply by nikolasburk** [reply]
> I tried to create relations but they always forced me to use foreign key or unique constraints but it's not possible here.

Yeah, FKs are a requirement for Prisma's relations, so I guess this is a limitation that's probably only possible to be worked around with raw SQL... It would be interesting to learn more about your specific use case here though, would you mind [opening a feature request](https://github.com/prisma/prisma/issues/new/choose) that describes your use case for relations that are not based on FKs so that we can evaluated this?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
